src/include/simeng/OS/SimOS.hh

```diff
@@ -63,11 +74,27 @@ class SimOS {
   /** Check if OS has halted. */
   bool hasHalted() const { return halted_; };
 
+  /** Method which allocates page frames of the specific size and returns the
+   * starting physical address. */
+  uint64_t requestPageFrames(size_t size);
+
+  /** Method which handles process specific page table translation. */
+  uint64_t handleVAddrTranslation(uint64_t vaddr, uint64_t tid);
+
+  /** Unique pointer to host backed file mmap. */
+  std::unique_ptr<HostBackedFileMMaps> const hfmmap_ =
+      std::make_unique<HostBackedFileMMaps>();
+
   /** Set up friend class with RegressionTest to enable exclusive access to
    * private functions. */
   friend class ::RegressionTest;
 
  private:
+  /** Function used to send data to memory without any timing constraints
+   * applied. This function is passed to the PageTable::handlePageFault when a
```

Comment:
"passed to the PageTable::handlePageFault" is inaccurate. It is passed to Process::handlePageFault

Reply:
done